asciidoc: Add to hugo-build, remove from install [semver:minor]

justwheel opened the pull request:
Closes #42.

### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [x] Examples have been added for any significant new features
- [x] README has been updated, if necessary

### Motivation, issues

This commit fixes a broken implementation introduced in commit ac08b667d45e2a57d615d24a2a260fded6627337. 

### Description

AsciiDoc was mistakenly added to the Hugo installation step, when it actually needs to be available at run-time, not build-time. This moves the AsciiDoc check logic into the Hugo run-time part of the pipeline so AsciiDoc is available in the `$PATH` if needed.

justwheel committed:
:beetle: :wrench: asciidoc: Add to hugo-build, remove from install

Closes #42.

This commit fixes a broken implementation introduced in commit
ac08b667d45e2a57d615d24a2a260fded6627337. AsciiDoc was mistakenly added
to the Hugo installation step, when it actually needs to be available at
run-time, not build-time. This moves the AsciiDoc check logic into the
Hugo run-time part of the pipeline so AsciiDoc is available in the
`$PATH` if needed.

Signed-off-by: Justin W. Flory (he/him) [UNICEF Innovation] <jflory@unicef.org>

justwheel commented:
The test suite is still running. I'm not 100% confident on this implementation, but it should work. A close review for best practices is appreciated.

dsayling commented on src/commands/install.yml:
```diff
@@ -7,10 +7,6 @@ parameters:
     description: "Install the Hugo extended binary?"
     type: boolean
     default: true
-  asciidoc:
```
I would go ahead and leave this here since you added the functionality. If a user is just using the install command and nothing else in the orb, they would lose this ability now.

justwheel replied:
@dsayling I think it is misleading in my opinion. The reason my CI pipeline fails is because `hugo/build` step is missing `/usr/bin/asciidoc` at run-time. Installing the `asciidoc` package in the `install` step works, but has no actual impact to building or compiling Hugo. I believe it adds an unnecessary step in the `install` step and might mislead someone who thinks it does something that it doesn't.

What do you think?

dsayling replied:
Totally agree - I'm more concerned about the technical API change. A removal of this parameter would technically necessitate a major version bump, however, I'm not sure that's something we should do in this case. Hopefully @KyleTryon can provide a little direction.

justwheel replied:
Understood! I'll resolve this thread and we can continue in the main thread.

dsayling commented on src/commands/hugo-build.yml:
```diff
@@ -16,7 +16,23 @@ parameters:
     description: "Additional flags to pass when running Hugo (e.g. -DF)."
     type: string
     default: ""
+  asciidoc:
```
If you want to use the orb job, you'll still need to update `src/jobs/build.yml` to accept the param and add it to the step where this command is called: https://github.com/CircleCI-Public/hugo-orb/blob/master/src/jobs/build.yml#L46

justwheel replied:
@dsayling Thanks for pointing this out. Resolved in commit 6266764a8f74efa9d4d9bed152405b42589bc93d.

justwheel committed:
:wrench: Integrate `asciidoc` parameter into build.yml config

This makes sure the `asciidoc` bool is available in the Orb and will
handle user configuration to install AsciiDoc. Still untested as of yet.

Signed-off-by: Justin W. Flory (he/him) [UNICEF Innovation] <jflory@unicef.org>

justwheel commented:
I updated the `build.yml` to include the new parameter. However, I still feel like the `asciidoc` step is incorrectly implemented in the `install` step. Let me know your thoughts on the above, or if this is ready for a merge.

dsayling commented:
@KyleTryon Should we remove the install parameter that was added incorrectly and do a major release, or could we get away with a minor release?

justwheel commented:
A minor release is preferable to me. Realistically I am probably the only one making use of this new feature right now. And we are within ~72 hours of when #40 was merged. To me, a major release would indicate something larger having changed, but that is not really the case here.

This is my 2¢, but either way is fine. Looking forward to trying this in my pipeline! 🚀

KyleTryon commented:
> @KyleTryon Should we remove the install parameter that was added incorrectly and do a major release, or could we get away with a minor release?

This could technically be considered a bug fix in a sense, especially given the quick time turn around but I'm much more comfortable calling it a minor release
